ReactViewControl.Avalonia/ReactViewRender.Avalonia.cs

```diff
@@ -46,6 +49,14 @@ partial class ReactViewRender : Control {
             }
         }
 
+        private static void OnHiddenWindowClosing(object sender, CancelEventArgs e) {
+            e.Cancel |= ((IClassicDesktopStyleApplicationLifetime)Application.Current.ApplicationLifetime).Windows.Count() != 0;
```

Comment:
if this window is not the last to be closing... this will not work as pretended

Reply:
hidden windows are always the last in the chain. Remember I tried that idea with sentrie window last week - no matter when it was created/opened. If it is hidden - it will always trigger last on CMD + Q. Moreover, its OnClosing event doesn't even fire if we close application in a standard way. I was only able to hit this event only on CMD + Q

Reply:
what about on windows? is it true also?

Reply:
on windows we don't have analogy for CMD + Q and the regular closing worked fine there as well when I ran it, but I didn't check if it hit that event or not. Will double check everything and come back with results

Reply:
yeah, it seems that it triggers only on CMD + Q, because all windows objects listen that event. I wasn't able to find a way to trigger it using something else, both on Win and Mac.